src/app/asset-page/asset-page.component.ts

```diff
@@ -765,15 +765,15 @@ export class AssetPage implements OnInit, OnDestroy {
         else {
             for (let col of this.collections) {
                 // Private/Personal Collection
-                if (col.type === '6') {
+                if (col.type === '6' && col.name === value) {
                     return ['/pcollection', col.id]
                 }
                 // Public Collection
-                else if (col.type === '5') {
+                else if (col.type === '5' && col.name === value) {
```

Comment:
Additional observation: 
This conditional is sort of a duplicate of the else, except this clause is really only used to set the `publicDownload` variable.